### ISSUE TITLE
Should redirect to the same page after signup

### ISSUE THREAD

**Issue by SidharthBansal** [publiclab/plots2#3367]
### Please describe the problem (or idea)
A person is on the /wiki and then signs up. The person is redirected to the dashboard

> What did you expect to see that you didn't?

I should be redirected to the wiki page after log in.

Hint use return_to field

**Comment by geekychaser** [comment]
May I take this?

**Comment by SidharthBansal** [comment]
You are really quick rishabh....
Why not???
Take it...If you need help just ping any of us at pl.
Thanks for contributing here at pl.

**Comment by geekychaser** [comment]
OK. Working on it.

**Comment by avsingh999** [comment]
It's redirecting after login to same  wiki page

**Comment by SidharthBansal** [comment]
@avsingh999 thanks for your help
I got this error when I was initially at /wiki and then I logged in. 
I was redirected to dashboard.

I am not talking of /wiki/_____. That issue was solved by me in month of feb or earlier.

**Comment by SidharthBansal** [comment]
I checked it. I am facing this error. @geekychaser can you please check whether this is arising at your machine?
Thanks all

**Comment by SidharthBansal** [comment]
@Uzay-G

**Comment by SidharthBansal** [comment]
Please ensure that this is working after your pr

**Comment by Uzay-G** [comment]
Yeah this works as you can see here:
![wiki-redirect](https://user-images.githubusercontent.com/52892257/71873561-c5c52400-311f-11ea-9853-fdb36c4e2dd0.gif)

**Comment by SidharthBansal** [comment]
Kindly work on some related stuff or new issue. I will be very less
available till monday. I have some important work.
I will review this whenever I get time.
You can abandon current task and get approval for mk and swb installation.
Please do other tasks after that. I will keep review login related task but
at slower pace. Sorry for inconvenience.

**Comment by SidharthBansal** [comment]
Once we will complete this task then you can get points for this.

**Comment by Uzay-G** [comment]
Alright no problem. I will abandon it and then work on some other tasks!
Thanks for your help!

**Comment by SidharthBansal** [comment]
Sure thanks

**Comment by SidharthBansal** [comment]
https://github.com/publiclab/plots2/issues/3367#issuecomment-571455615 I think you are missing something.

Fix in #7113 is for LOGIN

THis issue is specific to SIGN UP

**Comment by SidharthBansal** [comment]
We will need a pr for this too

**Comment by SidharthBansal** [comment]
This will be a hard task for GCI

**Comment by Uzay-G** [comment]
I think this works with the pull request I made, I just am struggling to link oauth as admin because I only have one github account to link to.

**Comment by SidharthBansal** [comment]
Hi

**Comment by SidharthBansal** [comment]
Sorry for misunderstanding, 
let me clarify you one thing. You need to show DASHBOARD AFTER SIGN UP THROUGH OAUTH
This is the desired functionality.

**Comment by SidharthBansal** [comment]
Please focus on last comment only.
Do following things in order
1. Install MK
2. Install SWB
3. Get points for both 1 and 2
4. Set up developer account for Github by link I provided in one of the issues
5. Try logging in to localhost with Github account.
Do this much first.

**Comment by Uzay-G** [comment]
Ok, sorry for misunderstanding, no need to get mad.
I already did 1-3 and am working on the last 2. I am sorry if I am bothering you with my questions.

**Comment by SidharthBansal** [comment]
6. run MK on localhost 3002
7. run SWB on localhost 3004
8. run Plots2 on localhost 3000
9. change https://github.com/publiclab/mapknitter/blob/5f25aa60243918b9d6a8f50a3db862358b25dafa/app/controllers/sessions_controller.rb#L16 to localhost:3000 like thingy. [Important: we don't want step 9 to reflect in any PR]
10. Try logging in MK through Plots2 form based authentication
11. Try logging in SWB through Plots2 form based authentication
12.  Try logging in MK through Plots2 GIthub based authentication
13. Try logging in SWB through Plots2 GIthub based authentication
do this with and without your changes.

**Comment by SidharthBansal** [comment]
Even if you are able to set up provider and connect SWB & MK with plots2 you will get 5 points.

**Comment by SidharthBansal** [comment]
Refer https://github.com/publiclab/plots2/blob/master/doc/LOGIN_SYSTEMS.md for help

**Comment by Uzay-G** [comment]
Hey, I managed to get the github app working and can sign up through it with plots2 but I cannot login with Mapknitter. Could you help me?

**Comment by SidharthBansal** [comment]
Please refer to above steps and tell me where are you having problem?

**Comment by Uzay-G** [comment]
Yeah I did and am having problems at step 10. This is the error message:
![error-auth](https://user-images.githubusercontent.com/52892257/71917247-eae97f00-317f-11ea-8217-51917cd4b7c6.png)

**Comment by SidharthBansal** [comment]
Use `@openid_url_base = "https://localhost:3000/people/"`
If that will not work then  `@openid_url_base = "localhost:3000/people/"`

**Comment by SidharthBansal** [comment]
for first statement to work use 3002 port for plots2, 3004 for MK and 3006 for SWB
3002 is for https plots2 while 3000 is for http plots2 version. I hope you understood this. Please ask if any doubts!

**Comment by Uzay-G** [comment]
Yeah I understood. It worked! Thanks :+1:

**Comment by SidharthBansal** [comment]
provide me gif so that we can move forward. It's night here. So, I can
sleep anytime... lol.
Will reply you till I am awake tonight

**Comment by SidharthBansal** [comment]
Gif for logins from Github and form at MK 5 points
Similarly for SWB
5 points for tests of SWB and MK combined if needed
so in total cross domain authentication carries 10 OR 15 points. I hope you
are okay with the marking scheme. I will upload things after we are done
with merging. Please remind me on 18 january to approve all of these Auth
related tasks.
Thanks

**Comment by Uzay-G** [comment]
Here is the gif for mapknitter but the redirection does not work yet:
![mapknitter-oauth](https://user-images.githubusercontent.com/52892257/71918522-bc20d800-3182-11ea-8362-9b6d54fdab27.gif)

**Comment by Uzay-G** [comment]
I will fix it soon!

**Comment by Uzay-G** [comment]
Thanks for all your help and don't worry I can figure things out if you need to go!

**Comment by SidharthBansal** [comment]
Hello,
One mistake you did in hurry. Kindly think carefully.
Once you logged in MK with plots2 icon or Github. It was the same thing for
MK
You should sign out the Plots2 also for both flows.
Let's stick to MK first. We will see SWB once MK pr is merged. Thanks

**Comment by SidharthBansal** [comment]
Kindly sign out github too before staring both the flows

**Comment by SidharthBansal** [comment]
starting**
I am sleepy.

**Comment by Uzay-G** [comment]
I will sign out of Mapknitter and send another gif but the redirection for oauth will only work when I add some edits to the views. Do you want me to do them now?

**Comment by SidharthBansal** [comment]
Send me two gifs by remaining at master of both mk and plots2. Plots2
logged out, GitHub logged out and mK logged out for both flows.
1. Login with mk and tell me what happens in gif
2. Logout mk logout plots2 and logout GitHub. Login mk by clicking on
GitHub icon. Provide me screenshot

Then we will move forward.

**Comment by SidharthBansal** [comment]
Master means no edits based branch

**Comment by Uzay-G** [comment]
Ok, I am doing that and login of MK is only working with oauth because Plots2 says I have not set a password although I have several times. I will show gif

**Comment by Uzay-G** [comment]
Functional login through Github: 
![mapknitter-oauth gif](https://user-images.githubusercontent.com/52892257/71919877-f63fa900-3185-11ea-91a7-4719951b61d0.gif)

**Comment by Uzay-G** [comment]
But for some reason I cannot normal login through plots2 or MK. I get the error: `This account doesn't have a password set. It may be logged in with a Github account, or you can set a new password via Forget password feature` although I have reset the password.

**Comment by SidharthBansal** [comment]
Please show me screenshots for this error

**Comment by Uzay-G** [comment]
Just one second I think I can fix it. Don't worry I will solve this and I know it's late where you are. You should go to sleep and I'll do it!

**Comment by SidharthBansal** [comment]
Cool gn

**Comment by Uzay-G** [comment]
Alright it works now:
Login through website:
![mapknitter-auth](https://user-images.githubusercontent.com/52892257/71921201-b0d0ab00-3188-11ea-844b-510a1fb0376f.gif)
Through Github:
![mapknitter-oauth](https://user-images.githubusercontent.com/52892257/71921329-ef666580-3188-11ea-9e2e-466fe638f0b2.gif)

**Comment by Uzay-G** [comment]
Now that that works I will setup redirection for that mapknitter on my own.

**Comment by Uzay-G** [comment]
Actually, the redirection is already working perfectly for mapknitter.

**Comment by Uzay-G** [comment]
Also everything is good with Spectral workbench:
Normal Login:
![spectral-auth](https://user-images.githubusercontent.com/52892257/71922084-8d0e6480-318a-11ea-9f6d-92b0f8a6aea9.gif)

Through Github:
![spectral-oauth](https://user-images.githubusercontent.com/52892257/71922097-93044580-318a-11ea-950a-d46d8753165e.gif)

**Comment by SidharthBansal** [comment]
Are these gifs with master or with your plots2 or changes?

**Comment by SidharthBansal** [comment]
Congrats you earned 10 pts :tada:
We need tests now. One question above

**Comment by Uzay-G** [comment]
These are with my changes but they don't affect this part of the interface. My changes only affected plots2. I will work on the tests!
Did you manage to sleep? I hope I didn't bother you and thanks again for your help

**Comment by Uzay-G** [comment]
I'll make tests for mapknitter and spectral!

**Comment by SidharthBansal** [comment]
Please provide above 2 gifs without code changes in plots2

**Comment by SidharthBansal** [comment]
Focus on MK first. You just need to copy and paste things for SWB

**Comment by SidharthBansal** [comment]
I think(not sure) you just need to write tests at plots2. You need to check
this thoroughly. Maybe all three repos(mk+swb+plots2) or only one
repo(plots) or 2 repos(mk+swb)
In case of doubts, feel free to contact me. I will be in airport from 10am
IST today to 8 pm IST. So, will be less available.

**Comment by Uzay-G** [comment]
This is the result for Mapknitter without my changes:
Normal login:
![mapknitter-auth](https://user-images.githubusercontent.com/52892257/71956813-65e87f00-31ec-11ea-8f90-ac2d0b2a4b8e.gif)

Github login:
![mapknitter-oauth](https://user-images.githubusercontent.com/52892257/71956829-6ed95080-31ec-11ea-8144-549a9039bf0a.gif)

**Comment by SidharthBansal** [comment]
This is strange, things are working on localhost correctly but not working
on production branch.

**Comment by SidharthBansal** [comment]
Production environment**

**Comment by Uzay-G** [comment]
Yeah that's white.

**Comment by SidharthBansal** [comment]
Sorry I can't understand. My flight just landed

**Comment by Uzay-G** [comment]
Sorry I meant weird

**Comment by Uzay-G** [comment]
Are there any issues I need to fix with spectral or MK then?
For plots2,
I already inserted some oauth redirection tests in #7113 and #7072.

Another thing I will fix is that currently, when a user signs up through oauth, they are not redirected to the dashboard. When we merge #7113, I will open a new pull request to fix that.

**Comment by SidharthBansal** [comment]
Please ping me once you are done with all the assigned tasks. If anything
is not remaining then please let me know.
I will review things altogether.

**Comment by Uzay-G** [comment]
Yeah I already included the tests for plots2 and everything seems fine with MK and SWB. I will work on other tasks in the meantime. Have a nice vacation!

**Comment by Uzay-G** [comment]
I will make the other pull requests after so there are no conflicts :+1:

**Comment by SidharthBansal** [comment]
cool

**Comment by SidharthBansal** [comment]
No no. We have to do things in order. Kindly do those prs too.

**Comment by Uzay-G** [comment]
Yeah I will make the other pull request but we need to merge these ones first otherwise there will be conflicts between the prs.

**Comment by SidharthBansal** [comment]
try to add tests in the same pr then

**Comment by Uzay-G** [comment]
Well actually my changes are not tests but other fixes, I will include these fixes in the pull requests if you like!

**Comment by SidharthBansal** [comment]
what fixes?

**Comment by SidharthBansal** [comment]
Either open it as a new pr OR as a new single commit in current pr so that
your changes are visible

**Comment by Uzay-G** [comment]
I already put the tests in the prs but for example right now on PL, signing up through oauth always redirects to the dashboard. 

Since my changes make it return to the previous page instead, I need to make signup through oauth redirect to dashboard instead

**Comment by Uzay-G** [comment]
> Either open it as a new pr OR as a new single commit in current pr so that
your changes are visible

I will insert the changes in #7113

**Comment by SidharthBansal** [comment]
Hi I will review asap. Tired now. I hope you can understand.

**Comment by SidharthBansal** [comment]
> I will insert the changes in #7113

What have you inserted?

**Comment by SidharthBansal** [comment]
task related to this issue is https://codein.withgoogle.com/dashboard/task-instances/5266798652424192/

**Comment by SidharthBansal** [comment]
@Uzay-G is this issue resolved?

**Comment by Uzay-G** [comment]
Currently, the user is redirected to the same page after signpb. But @jywarren and others seemed to say it would be better to redirect to the dashboard when signing up. That's why I opened https://github.com/publiclab/plots2/issues/7211

**Comment by SidharthBansal** [comment]
correct. this issue is not needed any more
moved to #7211
